packages/jsii-spec/lib/spec.ts

```diff
@@ -330,29 +330,38 @@ export enum PrimitiveType {
 }
 
 /**
- * A reference to a type (primitive, collection or fqn).
+ * Represents some instance of a type, for example as the return value of a method, a parameter, or a property.
  */
-export type TypeReference = TypeReferenceBase & (NamedTypeReference | PrimitiveTypeReference | CollectionTypeReference | UnionTypeReference);
+export interface TypeInstance<T extends TypeReference> {
```

Comment:
I think I'm going to say that this interface ony makes sense for return types, and I think I object to its use in other places.